### PR TITLE
Institution by lei

### LINKED ISSUE #2135: Filing API - institutions by id
As part of the integration between the UI and the API it would be good to have the  [`/institutions/<institutionId>`](https://github.com/cfpb/hmda-platform/blob/8222f7e8e5f3089e87eb210fc977bec0be8b0d98/docs/v1/API.md#institutions-by-id) (links to v1 docs) endpoint to be able to remove [the mocked response from the UI](https://github.com/cfpb/hmda-platform-ui/blob/32b6b2baff8ed3ca945a2a01b8d71bf62378241d/src/api/api.js#L15-L69). This will help us to not have to worry about updating status codes manually when working through the process.

_[`/institutions`](https://github.com/cfpb/hmda-platform/blob/8222f7e8e5f3089e87eb210fc977bec0be8b0d98/docs/v1/API.md#institutions) is less important until authentication comes in. For now, we've [mocked `Bank 0` and `Bank 1`](https://github.com/cfpb/hmda-platform-ui/blob/32b6b2baff8ed3ca945a2a01b8d71bf62378241d/src/api/api.js#L3-L13)._

### PR DESCRIPTION
**Pull request by omnipresent07**
Fixes #2135 

- Changes `FilingPersistence` to notify `InstitutionPersistence` of a new Filing. 
- Creates `/institutions/<lei>` endpoint to return institution and all of its filings
- Protobuf/serialization

### CODE REVIEW

**Comment by nickgrippin** [comment]
Looks like you've got a lot of unused imports and variables when running `sbt clean compile`

**Comment by awolfe76** [comment]
Can we update [the API docs](https://github.com/cfpb/hmda-platform/blob/074749f418246e1d52db10e4d7d40ca4474b6a83/docs/v2/api/filing-api.md) for this one too?

File: docs/v2/api/filing-api.md [hunk 1]
```diff
@@ -41,6 +41,98 @@ Quick links:
...
+
+`GET` - Returns the institution by its `lei` and its filings
+
+Example response:
+
+```json
+{
+    "institution": {
+        "activityYear": 2014,
+        "LEI": "10Bx939c5543TqA1144M",
```
Comment (awolfe76) [inline docs/v2/api/filing-api.md]: Is this `LEI` or `lei`? For the mock endpoint (and UI code changes) we were using lowercase `lei`.

Comment (omnipresent07) [reply docs/v2/api/filing-api.md]: It is uppercase `LEI`.

Comment (awolfe76) [reply docs/v2/api/filing-api.md]: Would it be a problem to change it to lowercase?

Comment (omnipresent07) [reply docs/v2/api/filing-api.md]: Sure, we can make it lowercase. I'll send another PR to this. Just need to make sure the refactoring doesn't affect the test cases.

Comment (awolfe76) [reply docs/v2/api/filing-api.md]: Cool with me, 👍.

**Comment by nickgrippin** [comment]
The `InstitutionHttpApi` trait needs an `ErrorResponseCodec` import statement.  Currently a 404 response returns the nested URI structure instead of the path.

**Comment by jmarin** [comment]
Can you please check what's up with `Travis` before we merge? Thanks

**Comment by omnipresent07** [comment]
I kicked off a new build on `Travis` and it passes this time. The test not passing in the last build was in [`LarValidationHttpApiSpec.scala`](https://github.com/cfpb/hmda-platform/blob/master/hmda/src/test/scala/hmda/api/http/public/LarValidationHttpApiSpec.scala#L65)

**Pull request merged**